src/audio/buffer.c

```diff
@@ -40,6 +41,14 @@ struct comp_buffer *buffer_alloc(uint32_t size, uint32_t caps, uint32_t align)
 		return NULL;
 	}
 
+	buffer->lock = rzalloc(SOF_MEM_ZONE_RUNTIME, SOF_MEM_FLAG_SHARED,
+			       SOF_MEM_CAPS_RAM, sizeof(*buffer->lock));
```

Comment:
Lock being in a separate location from the buffer? This feels a bit sketchy, do we have a good reason to not move the buffer itself?